examples/example_10_synthetic_phantom_dead-zone_MLEM_vs_SIRT.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+"""
+This example reproduces the l2 dead-zone reconstruction of the phantom used in:
+
+- N. Viganò and V. A. Solé, “Physically corrected forward operators for
+induced emission tomography: a simulation study,” Meas. Sci. Technol., no.
+Advanced X-Ray Tomography, pp. 1–26, Nov. 2017.
+
+@author: Nicola VIGANÒ, Computational Imaging group, CWI, The Netherlands,
+and ESRF - The European Synchrotron, Grenoble, France
+"""
+
+import numpy as np
+from numpy.typing import ArrayLike
+from typing import Tuple
+
+import matplotlib.pyplot as plt
+
+import corrct as cct
+
+try:
+    import phantom
+except ImportError:
+    cct.testing.download_phantom()
+    import phantom
+
+
+def cm2inch(x: ArrayLike) -> Tuple[float, float]:
+    """Convert cm to inch.
+
+    Parameters
+    ----------
+    x : ArrayLike
+        Sizes in cm.
+
+    Returns
+    -------
+    Tuple[float, float]
+        Sizes in inch.
+    """
+    return tuple(np.array(x) / 2.54)
+
+
+vol_shape = [256, 256, 3]
+data_type = np.float32
+
+ph_or = np.squeeze(phantom.modified_shepp_logan(vol_shape).astype(data_type))
+ph_or = ph_or[:, :, 1]
+
+(ph, vol_att_in, vol_att_out) = cct.testing.phantom_assign_concentration(ph_or)
+(sino, angles, expected_ph, background_avg) = cct.testing.create_sino(
+    ph, 30, psf=None, add_poisson=True, dwell_time_s=1e-2, background_avg=1e-2, background_std=1e-4
+)
+
+bckgnd_weight = np.sqrt(background_avg / (vol_shape[0] * np.sqrt(2)))
+
+num_iterations_sirt = 10000
+num_iterations_mlem = 50
+lower_limit = 0
+vol_mask = cct.processing.circular_mask(ph_or.shape)
+
+sino_substract = sino - background_avg
+
+sino_variance = cct.processing.compute_variance_poisson(sino)
+sino_weights = cct.processing.compute_variance_weight(sino_variance)
+
+lowlim_l2 = cct.solvers.Constraint_LowerLimit(0, norm=cct.solvers.DataFidelity_l2())
+lowlim_l2w = cct.solvers.Constraint_LowerLimit(0, norm=cct.solvers.DataFidelity_wl2(1 / bckgnd_weight))
+
+data_term_ls = cct.solvers.DataFidelity_l2()
+data_term_lsw = cct.solvers.DataFidelity_wl2(sino_weights)
+data_term_lsb = cct.solvers.DataFidelity_l2b(sino_variance)
+
+regul_weight = 1
+regul = cct.regularizers.Regularizer_TV2D(regul_weight)
+
+
+with cct.projectors.ProjectorUncorrected(ph.shape, angles) as A:
+
+    solver_wls_l = cct.solvers.PDHG(verbose=True, data_term=data_term_lsw, regularizer=[lowlim_l2w])
+    rec_wls_l, _ = solver_wls_l(A, sino_substract, num_iterations_sirt, x_mask=vol_mask)
+    #
+    solver_sirt = cct.solvers.SIRT(verbose=True, data_term=data_term_lsw)
+    rec_sirt, _ = solver_sirt(A, sino_substract, num_iterations_sirt, x_mask=vol_mask)
+
+    solver_mlem = cct.solvers.MLEM(verbose=True)
+    rec_mlem, _ = solver_mlem(A, sino_substract, num_iterations_mlem, x_mask=vol_mask, lower_limit=1e-5, upper_limit=10)
```

Comment:
why is there an upper limit? Do we have some slight instabilities?
If yes, I suggest to have a look at some other implementation (e.g. the one from @astra-toolbox, to see how they handle it)

Reply:
I might have a hint for this, or at least something to make sure doesn't interfere with the reconstruction: The signal is background subtracted, but the information about the background is not included in the MLEM reconstruction.

I mean that the signal `yb` that is passed to the reconstruction, is actually `yb = P(y + b) - b`, where `b` is the background and `P(.)` is the Poisson statistical distribution function. MLEM might be expecting `b = 0`, and thus to have `P(y)` as input data. To be verified!

Reply:
indeed, running the simulation without background noise (`background_avg=0.` and `background_std=0.`) produces "normal" results. Masking or not does not impact the reconstruction in the particular test.
I remove the noise from the test.

Reply:
The background is now handled correctly